cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -3444,24 +3444,6 @@ pruneInstallPlanPass1 pkgs
             <> optionalStanzasWithDepsAvailable availablePkgs elab pkg
     addOptionalStanzas elab = elab
 
-    setDocumentation :: ElaboratedConfiguredPackage -> ElaboratedConfiguredPackage
-    setDocumentation elab@ElaboratedConfiguredPackage{elabPkgOrComp = ElabComponent comp} =
-      elab
-        { elabBuildHaddocks =
-            elabBuildHaddocks elab && documentationEnabled (compSolverName comp) elab
-        }
-      where
-        documentationEnabled c =
-          case c of
-            CD.ComponentLib -> const True
-            CD.ComponentSubLib _ -> elabHaddockInternal
-            CD.ComponentFLib _ -> elabHaddockForeignLibs
-            CD.ComponentExe _ -> elabHaddockExecutables
-            CD.ComponentTest _ -> elabHaddockTestSuites
-            CD.ComponentBench _ -> elabHaddockBenchmarks
-            CD.ComponentSetup -> const False
-    setDocumentation elab = elab
-
```

Comment:
Wow, this is surprising. Why is this no longer needed? Or was it spurious from the start?

A changelog file would be very welcome.

BTW, thank you for the `findDependencyInStore` improvement.

Reply:
I explained in the commit message

> The `elabBuildHaddocks` option is only consulted in `hasValidHaddockTargets`, at which point documentation building the executable component is disabled because `elabHaddockExecutables` is
`False`.

Reply:
I added a changelog entry.

Reply:
Apologies, I usually view not per-commit, but bulk. It's clear now.